sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubConnectionStringProperties.java

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs;
+
+import com.azure.core.amqp.implementation.ConnectionStringProperties;
+import com.azure.core.credential.TokenCredential;
+
+import java.net.URI;
+
+/**
+ * A utility class that parses a connection string into sections. A Event Hub connection string is a set of key value
+ * pairs separated by semi-colon. A typical example is
+ * {@code "Endpoint=sb://foo.EventHub.windows.net/;SharedAccessKeyName=someKeyName;SharedAccessKey=someKeyValue"}.
+ *
+ * <p>
+ * A connection may have the following sections:
+ * <ul>
+ *     <li>Endpoint, which is mandatory. The hostname part of it is the "Fully qualified namespace".</li>
+ *     <li>SharedAccessKeyName and SharedAccessKey, optional, used to authenticate the access to the EventHub.</li>
+ *     <li>SharedAccessSignature, optional, an alternative way to authenticate the access to the EventHub.</li>
+ *     <li>EntityPath, optional, the queue name or the topic name under the service namespace</li>
+ * </ul>
+ *
+ * <p>
+ * When you have a EventHub connection string, you can use {@link EventHubClientBuilder#connectionString(String)}
+ * to build a client. If you'd like to use a {@link TokenCredential} to access a Event Hub, you can use this utility
+ * class to take the fully qualified namespace and optionally the entity path (queue/topic name) from the connection
+ * string and then use {@link EventHubClientBuilder#credential(String, String, TokenCredential)}.
+ * </p>
+ *
+ * @see EventHubClientBuilder#connectionString(String)
+ */
+public final class EventHubConnectionStringProperties {
+    private final URI endpoint;
+    private final String entityPath;
+    private final String sharedAccessKeyName;
+    private final String sharedAccessKey;
+    private final String sharedAccessSignature;
+
+    private EventHubConnectionStringProperties(ConnectionStringProperties properties) {
+        this.endpoint = properties.getEndpoint();
+        this.entityPath = properties.getEntityPath();
+        this.sharedAccessKeyName = properties.getSharedAccessKeyName();
+        this.sharedAccessKey = properties.getSharedAccessKey();
+        this.sharedAccessSignature = properties.getSharedAccessSignature();
+    }
+    /**
+     * Parse a Event Hub connection string into an instance of this class.
```

Comment:
Please review uses of Event Hub to ensure the right product name is used.
You have an Event **Hubs** namespace connection string, which is the one that does not have `EntityPath`. And an Event **Hub** connection string, which has `EntityPath` in it.

Reply:
According to your comment, useless member variable removed.